lib/ansible/module_utils/k8s_common.py

```diff
@@ -16,17 +16,26 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-import copy
-import json
+from __future__ import absolute_import, division, print_function
+
 import os
+import re
+import copy
+import base64
+
+from keyword import kwlist
 
+from ansible.module_utils.six import iteritems
 from ansible.module_utils.basic import AnsibleModule
 
 try:
-    from openshift.helper.ansible import KubernetesAnsibleModuleHelper, ARG_ATTRIBUTES_BLACKLIST
+    from openshift.helper import PRIMITIVES
```

Comment:
Out of curiosity, what are the benefits of using the openshift restclient instead of the kubernetes client?

IIUC, the openshift client models are generated using OpenShift origin, which doesn't necessarily have the latest supported API by upstream `Kubernetes`. This has caused me some problems in the past where I've needed newer APIs that weren't supported by the openshift client. This is not to say that upstream kubernetes-client is better or worse but I'd like to find a way for us to be able to consume the latest released version of kubernetes. Am I missing something?

Reply:
The OpenShift client is built on top of the Kubernetes client, and adds all of the OpenShift specific APIs and models. It does mean that we're relying on the version of Kubernetes supported in the OpenShift client, which may not always be the latest and greatest.

I think part of the reason for the delay in getting the latest Kubernetes APIs into the OpenShift client is that we don't yet have a community helping us maintain the OpenShift client. My hope is that by making the Ansible a bit more user friendly, and by putting all the module code under the Ansible umbrella, we'll gain community interest in both the modules and the underlying client.

Reply:
@chouseknecht thanks a bunch, it makes sense to me.

FWIW, I'm happy to help with the maintenance of the openshift client too. I'd like to help keeping the ansible module and the openshift client aligned with upstream kubernetes.